fpx-lib/src/api/models/otel.rs

```diff
@@ -74,14 +75,16 @@ impl Span {
                     let parent_span_id = if span.parent_span_id.is_empty() {
                         None
                     } else {
-                        Some(hex::encode(span.parent_span_id))
+                        // .unwrap is safe because we literally encode them in the exact same line as hex
+                        Some(HexEncodedId::new(hex::encode(span.parent_span_id)).unwrap())
```

Comment:
We should implement a From<u8> for our Hex encoded string, so that we do not need do this encode manually and then do the unwrap (even if it is safe)